### PR TITLE
Move CSIVolumeFSGroupPolicy feature to GA

### PR DESCRIPTION
**Pull request by dobsonj**
This updates the feature gates to move CSIVolumeFSGroupPolicy from beta to GA in 1.23.

#### What type of PR is this?

/kind feature
/sig storage
/milestone v1.23

#### Which issue(s) this PR fixes:

Enhancement: https://github.com/kubernetes/enhancements/issues/1682

#### Does this PR introduce a user-facing change?

```release-note
The CSIVolumeFSGroupPolicy feature has moved from beta to GA.
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

- KEP: https://github.com/kubernetes/enhancements/blob/master/keps/sig-storage/1682-csi-driver-skip-permission/README.md
- Docs: https://kubernetes-csi.github.io/docs/support-fsgroup.html

/cc @gnufied @msau42 @xing-yang

### CODE REVIEW

File: pkg/features/kube_features.go [hunk 1]
```diff
@@ -835,7 +836,7 @@ var defaultKubernetesFeatureGates = map[featuregate.Feature]featuregate.FeatureS
 	CSIStorageCapacity:                             {Default: true, PreRelease: featuregate.Beta},
 	CSIServiceAccountToken:                         {Default: true, PreRelease: featuregate.GA, LockToDefault: true}, // remove in 1.23
 	GenericEphemeralVolume:                         {Default: true, PreRelease: featuregate.GA, LockToDefault: true}, // remove in 1.25
-	CSIVolumeFSGroupPolicy:                         {Default: true, PreRelease: featuregate.Beta},
+	CSIVolumeFSGroupPolicy:                         {Default: true, PreRelease: featuregate.GA, LockToDefault: true}, // remove in 1.24
```
Comment (msau42) [inline pkg/features/kube_features.go:839]: I believe the deprecation policy is 2 releases, so this should be 1.25

Comment (dobsonj) [reply pkg/features/kube_features.go]: Ah thank you, updated.

File: pkg/features/kube_features.go [hunk 2]
```diff
@@ -330,9 +330,10 @@ const (
 	// Disables the OpenStack Cinder in-tree driver.
 	InTreePluginOpenStackUnregister featuregate.Feature = "InTreePluginOpenStackUnregister"
 
-	// owner: @huffmanca
+	// owner: @huffmanca, @dobsonj
 	// alpha: v1.19
 	// beta: v1.20
+	// GA: v1.23
```
Comment (msau42) [inline pkg/features/kube_features.go:336]: Can you also update the API comments to remove the beta designation?

https://github.com/kubernetes/kubernetes/blob/731ae59918970b95ee8fb509d211c81e200725af/pkg/apis/storage/types.go#L286

Comment (dobsonj) [reply pkg/features/kube_features.go:336]: Good catch, updated.

File: pkg/apis/storage/types.go [hunk 3]
```diff
@@ -283,8 +283,8 @@ type CSIDriverSpec struct {
 	// Defines if the underlying volume supports changing ownership and
 	// permission of the volume before being mounted.
 	// Refer to the specific FSGroupPolicy values for additional details.
-	// This field is beta, and is only honored by servers
-	// that enable the CSIVolumeFSGroupPolicy feature gate.
+	// This field is only honored by servers that enable the
```
Comment (msau42) [inline pkg/apis/storage/types.go:286]: I think we can remove this line altogether as the feature is locked to on and can't be disabled.

Comment (msau42) [reply pkg/apis/storage/types.go:286]: Also update the versioned (v1, v1beta, etc) types.go as well.

**Review by thockin — COMMENTED** [review]
/approve

**Pull request merged**